controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+	//   - that are not empty by default, so empty fields in the base podSpec do not remove default values
+	//   - and there is not and existing CF that handles them
+	// Otherwise we just pass the base values along, relying on CFs to override things
+
+	base = base.DeepCopy()       // Defensive copy TODO is this needed?
+	current = current.DeepCopy() // Defensive copy TODO is this needed?
+	// affinity
+	if base.Affinity != nil {
+		return nil, NewConfigError("affinity")
+	}
+	base.Affinity = current.Affinity
+	// containers[*]
+	if len(base.Containers) > 1 {
+		return nil, NewConfigError("field containers must contain at most one item")
+
+	} else if len(base.Containers) == 1 {
+		baseContainer := &base.Containers[0]
+		// We have only a single container at the moment,
+		// TODO but in the future we should use the name field.
+		currentContainer := current.Containers[0]
+		factoryContainer := factory.Containers[0]
+
+		// containers[*].args
+		if len(baseContainer.Args) > 0 {
+			return nil, NewConfigError("containers[0].args")
+		}
+		baseContainer.Args = currentContainer.Args
+		// containers[*].command
+		if len(baseContainer.Command) > 0 {
+			return nil, NewConfigError("containers[0].command")
+		}
+		baseContainer.Command = currentContainer.Command
+		// containers[*].env
+		if len(baseContainer.Env) > 0 {
+			return nil, NewConfigError("containers[0].env")
+		}
+		baseContainer.Env = currentContainer.Env
+		// containers[*].image
+		if baseContainer.Image != "" {
+			return nil, NewConfigError("containers[0].image")
+		}
+		baseContainer.Image = currentContainer.Image
+		// containers[*].imagePullPolicy
+		if baseContainer.ImagePullPolicy != "" {
+			return nil, NewConfigError("containers[0].imagePullPolicy")
+		}
+		baseContainer.ImagePullPolicy = currentContainer.ImagePullPolicy
+		// containers[*].livenessProbe
+		if baseContainer.LivenessProbe == nil {
+			baseContainer.LivenessProbe = factoryContainer.LivenessProbe
+		}
+		// containers[*].name
+		if baseContainer.Name != "" {
+			return nil, NewConfigError("containers[0].name")
+		}
+		baseContainer.Name = factoryContainer.Name
+		// containers[*].ports
+		if len(baseContainer.Ports) > 0 {
+			return nil, NewConfigError("containers[0].ports")
+		}
+		baseContainer.Ports = currentContainer.Ports
+		// containers[*].readinessProbe
+		if baseContainer.ReadinessProbe == nil {
+			baseContainer.ReadinessProbe = factoryContainer.ReadinessProbe
+		}
+		// containers[*].resources
+		if len(baseContainer.Resources.Limits) == 0 &&
+			len(baseContainer.Resources.Requests) == 0 {
+			baseContainer.Resources = factoryContainer.Resources
+		}
+		// containers[*].workingDir
+		if baseContainer.WorkingDir != "" {
+			return nil, NewConfigError("containers[0].workingDir")
+		}
+		baseContainer.WorkingDir = currentContainer.WorkingDir
+	} else {
+		base.Containers = make([]ar.ApicurioRegistrySpecDeploymentPodSpecPreviewContainer, len(factory.Containers))
+		for i, c := range factory.Containers {
```

Comment:
I'm not sure to read this correctly:
are we using `container[0]` from the default Operator factory and appending any additional container coming from `podTemplate`?